Objects/moduleobject.c

```diff
@@ -866,6 +866,40 @@ _Py_module_getattro(PyModuleObject *m, PyObject *name)
     return _Py_module_getattro_impl(m, name, 0);
 }
 
+static int
+module_setattro(PyModuleObject *m, PyObject *name, PyObject *value)
+{
+    PyObject *setattr, *delattr;
+    assert(m->md_dict != NULL);
+    PyErr_Clear();
+    if (value == NULL) {
+        delattr = PyDict_GetItemWithError(m->md_dict, &_Py_ID(__delattr__));
```

Comment:
Can you try using PyDict_GetItemRef() instead? Same remark below. It avoids checking PyErr_Occurred().

Reply:
> It avoids checking PyErr_Occurred().

This still requires a test for ``delattr == -1``.  Why this is better?

BTW:
```
$ git grep '= *PyDict_GetItemWithError'|wc -l
116
$ git grep '= *PyDict_GetItemRef'|wc -l
2
```

Reply:
Calling PyErr_Occurred() is slower than not calling it. Yes, PyDict_GetItemRef() API is new. I added it.